jax/_src/nn/functions.py

```diff
@@ -344,9 +345,15 @@ def softmax(x: Array,
       when :code:`where` is not None.
   """
   x_max = jnp.max(x, axis, where=where, initial=initial, keepdims=True)
-  unnormalized = jnp.exp(x - lax.stop_gradient(x_max))
```

Comment:
We no longer need `lax.stop_gradient` here because we no longer differentiate through this implementation, instead relying on the `custom_jvp` rule defined below.